iradix.go

```diff
@@ -357,6 +379,60 @@ func (t *Txn) delete(parent, n *Node, search []byte) (*Node, *leafNode) {
 	return nc, leaf
 }
 
+// delete does a recursive deletion
+func (t *Txn) deletePrefix(parent, n *Node, search []byte) (*Node, int) {
+	// Check for key exhaustion
+	if len(search) == 0 {
+		// Remove the leaf node
+		nc := t.writeNode(n, true)
+		if n.isLeaf() {
+			nc.leaf = nil
+		}
+		subTreeSize := 0
```

Comment:
Addressed this, had to make one minor tweak to `trackChannelsAndCount` to prevent overcounting non-leaf nodes.